### PR TITLE
feat(contrib,engine): add release action + artifactory release plugin

### PR DESCRIPTION
**Pull request by sguiheux**
Do not merge it. Waiting to merge #5824  first

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16101.2 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16134.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✘  
  * Test UI ✔

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16134.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16134.2 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: contrib/integrations/artifactory/plugin-artifactory-release/main.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+	log.SetLogger(log.NewLogger(log.ERROR, os.Stdout))
+	distriClient, err := art.CreateDistributionClient(artifactoryURL, releaseToken)
+	if err != nil {
+		return fail("unable to create distribution client: %v", err)
+	}
+
+	// Promotion
+	artiClient, err := art.CreateArtifactoryClient(artifactoryURL, token)
+	if err != nil {
+		return fail("unable to create artifactory client")
```
Comment (yesnault) [inline contrib/integrations/artifactory/plugin-artifactory-release/main.go:101]:
```suggestion
		return fail("unable to create artifactory client: %v", err)
```

File: contrib/integrations/artifactory/plugin-artifactory-release/main.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+	artifactPromoted := make([]promotedArtifact, 0)
+	for _, r := range runResult {
+		rData, err := r.GetArtifactManager()
+		if err != nil {
+			return fail("unable to read result %s: %v", r.ID, err)
+		}
+		skip := true
+		for _, reg := range artRegs {
+			if reg.MatchString(rData.Name) {
+				skip = false
```
Comment (yesnault) [inline contrib/integrations/artifactory/plugin-artifactory-release/main.go:128]: break missing

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16147.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#16147.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

**Pull request merged**